pkg/gitserver/gitserver.go

```diff
@@ -184,12 +197,22 @@ func NewEnviromentConfig() (*Config, error) {
 		return nil, fmt.Errorf("only one of REQUIRE_SERVER_AUTH or REQUIRE_GIT_AUTH may be specified")
 	}
 
+	config.WatchBuildConfigs = os.Getenv("BUILDCONFIG_AUTOSTART") == "yes"
```

Comment:
aren't the two orthogonal?  
auto-linking:  we figure out which buildconfig this repo belongs to.  
autostart=we run a build for the BC associated with this repo(if any) when a push happens.

autostart is unlikely to work without autolinking, but could if a user did the right configuration on their own.

i'm less clear on why watching buildconfigs is tied to autostart being true.  i would have thought it would be tied to autolinking.